py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -404,6 +404,54 @@ def __call__(self, driver):
             return False
 
 
+def any_of(*expected_conditions):
+    """ An expectation that any of multiple expected conditions is true.
+    Equivalent to a logical 'OR'.
+    Returns results of the first matching condition, or False if none do. """
+    def any_of_condition(driver):
+        for expected_condition in expected_conditions:
+            try:
+                result = expected_condition(driver)
+                if result:
+                    return result
+            except WebDriverException:
+                pass
+        return False
+    return any_of_condition
+
+
+def all_of(*expected_conditions):
+    """ An expectation that all of multiple expected conditions is true.
+    Equivalent to a logical 'AND'.
+    Returns a Boolean """
+    def all_of_condition(driver):
+        for expected_condition in expected_conditions:
+            try:
+                result = expected_condition(driver)
+                if not result:
+                    return False
+                return True
```

Comment:
`return True` should be outside the for loop.
Many expected conditions sometimes return the element or object you're waiting for.
Should this return a list of all results?

Reply:
#1 (facepalm) fixed
#2 That's a good idea.  the `all_of` function returns every EC's result in a list; it could then be parsed for elemental results if needed.